tests/test_002-exclusive_jets.py

```diff
@@ -701,3 +701,49 @@ def test_listoffset_indexed_input():
     ]
 
     assert inclusive_jets == cluster.inclusive_jets().to_list()
+
+
+def test_njettiness():
+    array = ak.Array(
+        [
+            [
+                {"px": 1.2, "py": 3.2, "pz": 5.4, "E": 2.5, "ex": 0.78},
+                {"px": 1.25, "py": 3.15, "pz": 5.4, "E": 2.4, "ex": 0.78},
+                {"px": 1.4, "py": 3.15, "pz": 5.4, "E": 2.0, "ex": 0.78},
+                {"px": 32.2, "py": 64.21, "pz": 543.34, "E": 24.12, "ex": 0.35},
+                {"px": 32.45, "py": 63.21, "pz": 543.14, "E": 24.56, "ex": 0.0},
+            ],
+            [
+                {"px": 1.2, "py": 3.2, "pz": 5.4, "E": 2.5, "ex": 0.78},
+                {"px": 1.25, "py": 3.15, "pz": 5.4, "E": 2.4, "ex": 0.78},
+                {"px": 1.4, "py": 3.15, "pz": 5.4, "E": 2.0, "ex": 0.78},
+                {"px": 32.2, "py": 64.21, "pz": 543.34, "E": 24.12, "ex": 0.35},
+                {"px": 32.45, "py": 63.21, "pz": 543.14, "E": 24.56, "ex": 0.0},
+            ],
+        ],
+        with_name="Momentum4D",
+    )
+
+    jetdef = fastjet.JetDefinition(fastjet.antikt_algorithm, 0.8)
+    jet_cluster = fastjet._pyjet.AwkwardClusterSequence(array, jetdef)
+
+    constituents = jet_cluster.constituents()
+
+    njettiness_cluster = fastjet._pyjet.AwkwardClusterSequence(constituents, jetdef)
```

Comment:
why does `AwkwardClusterSequence()` needs to be called twice?

Reply:
The output_wrapper (that contains the input to the jet clustering algorithm in FastJet components) doesn't look like it gets updated between calls, so we're stuck with this funky construction.

i.e. I'm pretty sure I can't do `(jet_cluster.njettiness(jet_cluster.constituents())`

I'll see if I can make it nicer, I agree it's sorta odd. But right now we have some strange internal state to deal with.

I'll poke at it a little to see if something more pleasant can be achieved. OTOH making another ClusterSequence costs nothing.

Reply:
could the `constituents()` and/or `AwkwardClusterSequence()` calls be made inside `njettiness()`?

Reply:
that's how I had it before! the way it's setup now allows us to use an arbitrary jet algorithm before Njettiness.

`.constituents()` implies inclusive, there's a separate constituents for exclusive jets. I will admit it is clunky but it's also outside the scope of this PR to fix it.

I'd vote in favor of flexibility of initial jet clustering algorithm.

Reply:
> there's a separate constituents for exclusive jets

that was the piece of information I was missing.

Reply:
Yeah, indeed, have to make two cluster sequences to set the input arrays to the constituents. It's weird but I think it's an OK price to pay for not having a new function per jet algorithm variant.

Reply:
What about adding an optional `cluster` argument in `constituents()` that, when true, calls `AwkwardClusterSequence()` internally? (and the same could be done for the exclusive jet version)

Reply:
Then we're back to needing a function per jet type, no?
I find that harder to maintain in the long run.

Reply:
No, I really just mean eliminating one line of code here.

In `_pyjet.py`:
```python
    def constituents(self, min_pt=0, cluster=False):
        result = self._internalrep.constituents(min_pt)
        if cluster:
            return AwkwardClusterSequence(result, self._jetdef)
        else:
            return result
```
then in the test (i.e. in whatever code users will write to use the feature):
```python
constituents_cluster = jet_cluster.constituents(cluster=True)
result = constituents_cluster.njettiness()
```
and a similar pattern can be implemented for the exclusive jet constituents.

Reply:
Ah, gotcha, this will require some care with the dask implementation since we have to avoid accidentally sending the client side cluster sequence to a node. It can work.

I would be in favor of this, but it's more substantial than the base scope of this PR. I would take this to an issue and we fix it with a different PR more consistently (this anti-pattern we're unhappy with crops up in other places).

Reply:
Agreed, opened #337.

Reply:
Sorry, why doesn't `njettiness` work directly on the first cluster here? i.e. 

```python
jet_cluster = fastjet._pyjet.AwkwardClusterSequence(array, jetdef)
njetti = jet_cluster.njettiness()
```

I'm confused because that seems to work on the "realistic" test below without calling `constituents()` first?

Reply:
If we called njettiness on the input data it would do the forced exclusive clustering you do in njettiness to the whole event!

Reply:
So you have to cluster jets first then do njettiness on the constituents of the jets l.

Reply:
Ah so that would give the event `njettiness` rather than per-jet `nsubjettiness`? OK then, LGTM, thanks!